### DISCUSSION TITLE
One-click update

### DISCUSSION THREAD

**Discussion by Bubbelwubbel** [pelican-dev/panel#262] [Ideas]
That you can update the panel or Wings in the panel with one click. This then executes the update script and if it goes well, the new update is immediately available.

**Comment by Googletod12355** [comment]
Would be good for this to be on the dashboard and to have a toggle for automatic updates and maby make it so the update only runs during a set time frame.

**Comment by TekExplorer** [comment]
I think this title is misleading.
Automatic updates implies that an update can occur without you doing anything, which is dangerous and can break something and you won't know why.

What I believe you mean, however, is one-click updates.

Honestly, I'm not sure how you'd do that when the panel is running in a container, as is the intended default for pelican.

Regardless, updating the panel once it's dockerized will be as easy as using `latest` (or manually updating the version) and running a docker compose pull and docker compose up -d

It's what I do with pterodactyl

**Comment by benzon** [comment]
I would be possible to not secure, since updating would require root permissions to update, so yea i dont think this is a good idea in any way.

Better to do a cron script that runs every 12/24 hour and does the update flow, but then your install flow needs to fit official flow 100% any ways.
